v2/internal/controllers/crd_networking_privatednszone_test.go

```diff
@@ -48,3 +53,48 @@ func Test_Networking_PrivateDnsZone_CRUD(t *testing.T) {
 	tc.Expect(retryAfter).To(BeZero())
 	tc.Expect(exists).To(BeFalse())
 }
+
+func VirtualNetworkLinks_CRUD(tc *testcommon.KubePerTestContext, zone *network.PrivateDnsZone, rg *resources.ResourceGroup) {
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+	tc.CreateResourceAndWait(vnet)
+
+	links := newVirtualNetworkLink(tc, zone, vnet)
```

Comment:
minor: It's just 1 link, right?
```suggestion
	link := newVirtualNetworkLink(tc, zone, vnet)
```